src/main/java/org/traccar/api/resource/UserResource.java

```diff
@@ -65,26 +67,22 @@ public UserResource() {
     @GET
     public Collection<User> get(
             @QueryParam("userId") long userId, @QueryParam("deviceId") long deviceId) throws StorageException {
+        var conditions = new LinkedList<Condition>();
         if (userId > 0) {
             permissionsService.checkUser(getUserId(), userId);
-            return storage.getObjects(baseClass, new Request(
-                    new Columns.All(),
-                    new Condition.Permission(User.class, userId, ManagedUser.class).excludeGroups()));
-        } else if (deviceId > 0) {
+            conditions.add(new Condition.Permission(User.class, userId, ManagedUser.class).excludeGroups());
+        } else if (permissionsService.notAdmin(getUserId())) {
+            conditions.add(new Condition.Permission(User.class, getUserId(), ManagedUser.class).excludeGroups());
+        }
+        if (deviceId > 0) {
             permissionsService.checkManager(getUserId());
-            var conditions = new LinkedList<Condition>();
             conditions.add(new Condition.Permission(User.class, Device.class, deviceId).excludeGroups());
             if (permissionsService.notAdmin(getUserId())) {
                 conditions.add(new Condition.Permission(User.class, getUserId(), ManagedUser.class).excludeGroups());
             }
-            return storage.getObjects(baseClass, new Request(new Columns.All(), Condition.merge(conditions)));
-        } else if (permissionsService.notAdmin(getUserId())) {
-            return storage.getObjects(baseClass, new Request(
-                    new Columns.All(),
-                    new Condition.Permission(User.class, getUserId(), ManagedUser.class).excludeGroups()));
-        } else {
-            return storage.getObjects(baseClass, new Request(new Columns.All()));
         }
+        return storage.getObjects(baseClass,
+                new Request(new Columns.All(), Condition.merge(conditions), new Order("name")));
```

Comment:
What's the change here?

Reply:
I think this comment is still not addressed.

Reply:
Refatorou o código para simplificar o tratamento de permissões centralizando todas as condições em uma única lista (`conditions`). Agora, as permissões são acumuladas nessa lista, e `storage.getObjects` é chamado apenas uma vez com todas as condições aplicadas.

Por exemplo, a verificação `if (permissionsService.notAdmin(getUserId()))` era chamada duas vezes em diferentes ramificações. Agora, ela está consolidada em uma única condição adicionada a `conditions`, reduzindo a redundância e melhorando a legibilidade.

**Observação:** O código mencionado está desatualizado, pois ainda contém a verificação duplicada para `permissionsService.notAdmin(getUserId())`.